### PR TITLE
[WFCORE-3963] Fix of WFCORE-3826 breaks plain authentic…

### PR DESCRIPTION
**Pull request by JiriOndrusek**
…ation for ejbs using legacy configuration

Issue: https://issues.jboss.org/browse/WFCORE-3963

Test: https://github.com/wildfly/wildfly/pull/11432

### CODE REVIEW

**Comment by xstefank** [comment]
@JiriOndrusek [WFCORE-3963] is mentioned twice in commit and PR title messages

**Comment by JiriOndrusek** [comment]
@xstefank Bad ctrl+c, I'll fix it, thanks

**Comment by kabir** [comment]
@darranl @fjuma please approve

File: domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmService.java [hunk 1]
```diff
@@ -590,7 +590,7 @@ public SaslAuthenticationFactory getSaslAuthenticationFactory(final String[] mec
 
             for(Iterator<String> iter = requestedMechanisms.iterator();iter.hasNext();) {
                 AuthMechanism authMechanism = toAuthMechanism("SASL", iter.next());
-                if(authMechanism != null) {
+                if(authMechanism != null && (registeredServices.containsKey(authMechanism) || registeredServices.containsKey(AuthMechanism.DIGEST))) {
```
Comment (fjuma) [inline domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmService.java]: @JiriOndrusek Why is there a special case for DIGEST here and below? Could you explain the reason for these changes a bit? Thanks!

Comment (JiriOndrusek) [reply domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmService.java]: Now I understand it correctly, (I hope):
- fix should allow anonymous - no problem
- fix should allow only these mechanisms, which are currently enabled by elytron, with exception for PLAIN which could use DIGEST (this substitution works, I've tested it)

@darranl  Is my understanding correct?

**Comment by JiriOndrusek** [comment]
@fjuma I'll change fix according to result of discussion:
- fix should allow only these mechanisms, which are currently enabled by elytron, with exception for PLAIN which could use DIGEST

**Comment by JiriOndrusek** [comment]
@fjuma changes are commited

**Pull request merged**